0001-typedb/01-start/04-iam-schema.md

```diff
@@ -0,0 +1,302 @@
+summary: Explanation of the IAM schema, widely used throughout TypeDB documentation.
+toc: false
+---
+
+# IAM schema explanation
+
+## Introduction
+
+**Identity and access management** (**IAM**) is a complex field that has seen massive growth over the past two decades 
+to meet user demand. IAM systems have become progressively more complicated to account for the variety of user 
+requirements and to ensure compatibility with a backlog of legacy systems, leading to the definition of standards by 
+governing bodies, such as [NIST](https://doi.org/10.6028/NIST.SP.800-63-3), in an attempt to unify and simplify the 
+state of IAM ecosystems, as well as to set out best practices. Modern IAM systems, such as the 
+[RSA’s Identity Governance and Administration Platform](https://www.rsa.com/solutions/identity-governance-and-administration/), 
+implement these models while providing APIs to existing IAM systems an organization might have in order to unify IAM 
+under a single management framework.
+
+<div class="note">
+[Note]
+The terminology used in the IAM [schema](03-quickstart.md#prepare-a-tql-file) is defined according to the sources 
+mentioned above and ISO/IEC 15408-1 standard within the context of an information system.
+</div>
+
+<div class="note">
+[Note]
+The Identity and Access Management database [schema](03-quickstart.md#prepare-a-tql-file) and accompanying 
+[dataset](03-quickstart.md#insert-data) described here are used in most examples throughout TypeDB documentation, 
+including the [Quickstart guide](03-quickstart.md).
+</div>
+
+## Overview
+
+As with any [TypeDB schema](../02-dev/02-schema.md), the IAM schema consists of:
+
+- Types
+  - Entity types
+  - Relation types
+  - Attribute types
+- Rules
+
+The [schema](../02-dev/02-schema.md) page will later provide more information to better understand the 
+terminology of TypeDB schemas.
+
+The main entity types in the IAM schema:
+
+- Subject — Active entity in the _system_ that performs operations on objects. 
+  - User — Human or IT entity possibly interacting with the _system_ from outside of the _system_ boundary.
+    - Person — User with `credentials`, `email`, and `full-name` attributes.
+  - User-group — A group of users that share the same role.
+- Object — Passive entity in the _system_, that contains or receives information, and upon which subjects perform
+  operations. 
+  - Resource — Anything usable or consumable in the _system_.
+    - File — File in some filesystem with `path` (to the file) and `size` (in KB) attributes.
+  - Resource-collection — A collection of resources on which the same operations can be performed.
+- Action — An operation or operation set that can be performed on a specific type of object. Its name is stored in
+  the `action-name` attribute.
+
+<div class="note">
+[Note]
+In the list above some entities have attributes that are mentioned in their description.
+</div>
+
+The main relation types in the IAM schema:
```

Comment:
I would also add memberships and ownerships to this list (if you use them at all).

Reply:
Why would we do that? They are not nearly as important as the two below.

Reply:
> Because permission inference is massively dependent on memberships. 

Not in my dataset. I don't use inference outside the single rule I have added. So, no membership is needed for documentation purposes. Whether we need to explain the whole schema for users — I don't think we need it right now. It will make our explanation page overcomplicated. And it's in the getting started section, so it needs to be as simple as possible.

Reply:
Yep that's fair enough. Didn't realise you don't use memberships.